frontend/src/scenes/session-recordings/player/snapshot-processing/process-all-snapshots.ts

```diff
@@ -95,8 +111,11 @@ export function processAllSnapshots(
                 }
             }
 
-            result.push(snapshot)
+            sourceResult.push(snapshot)
         }
+
+        processSnapshots.set(sourceKey, sourceResult)
+        result.push(...sourceResult)
```

Comment:
so now we're copying more
but we're storing the result for each source
and then not processing it again

it is way faster

(i need to de-risk the spread copying of potentially large arrays which we've had a problem with elsewhere